### DISCUSSION TITLE
Dealling with inadaquate DICOM

### DISCUSSION THREAD

**Discussion by MRIDude** [Donders-Institute/bidscoin#237] [General]
I am new to bidscoin. I been learning the software and testing it out on our data.  We have some old DICOMS from the mid 2000's that have impoverished headers. I learned this through Chris Rorden, it an issue with the scanner software. The DICOMS are for a duel-echo 2D tse data where dcm2niix labels the nifti files with the incorrect echo number.  Is there a way of dealing with these DICOMS. I have a feeling I need to set up these images outside of bidscoin.

**Comment by marcelzwiers** [comment]
If dcm2niix labels the files with the incorrect echonumber, but you know what the correct echo number is then BIDScoin may be able to use that and produce correct output. How is the correct echonumber stored?

**Reply by MRIDude** [reply]
Not sure what you mean by how is the correct echo number stored.   The echo number is in the DICOMS. Interestingly the first echo has it's echo number as a 0 in the DICOMS and the second echo is a 1. I always thought it should start with a 1 not 0. dcm2niix does not show the echo number in the json for the first echo but does show up in the second one. 

The conversation chris and I had can be found 
[https://github.com/rordenlab/dcm2niix/issues/821](url)

**Comment by marcelzwiers** [comment]
I see that the second echo comes out as echo 15. Well, if that's all then you could run bidscoin twice, the first time to get `echo-1` and `echo-15` images, and the second time with the nibabel plugin + bidsmap_bids2bids template. In the last step you can correct the echo numbers  (and put the correct echo number in the meta table).

**Reply by MRIDude** [reply]
Is there a tutorial for the nibabel plugin?

**Comment by marcelzwiers** [comment]
No, but I've helped a few people out in the past, so you may find some help in the posts/closed issues. Basically the nibabel plugin can convert anything that nibabel supports, but then you need to provide the meta data yourself (e.g. in the bidseditor). In case of nibabel plugin + nifti input data (converted by yourself, e.g. in a previous bidscoiner run), then you have to run the bidsmapper with the `-p nibabel2bids` plugin, and then you can correct the stuff that you want in the bidseditor. Then simply run the bidscoiner

**Reply by MRIDude** [reply]
Ya I already did that. I have spent the last week  and half trying to educate my self on this scripts going through all the issues. I have found some resources but not enough. We have decided to write a separate script to fix these errors.
